configure.ac

```
@@ -934,10 +934,67 @@ if test "x$enable_backtrace" = xunwind-ptrace; then
       )],
       [AC_DEFINE([HAVE_LIBUNWIND_ELF_FILENAME], [1], [Define if libunwind has unw_get_elf_filename])
       AC_MSG_RESULT(yes)],
-      [AC_MSG_RESULT(no)]
+      [AC_MSG_RESULT([You have selected libunwind, but this version does not support retrieving file names])]
    )
 fi
 
+AC_ARG_ENABLE([demangling],
+              [AS_HELP_STRING([--enable-demangling],
+                              [enable demangling support for backtraces @<:@default=check@:>@])],
+              [],
+              [enable_demangling=no])
+have_libiberty=no
+case $enable_demangling in
+   check|yes|libiberty)
+      AC_CHECK_LIB([iberty], [cplus_demangle], [
+         have_libiberty=yes
+         AC_DEFINE([HAVE_LIBIBERTY], [1], [Define to 1 if libiberty is found.])
+         LIBS="$LIBS -liberty"
+         ], [
+         if test "x$enable_demangling" = xlibiberty; then
+            AC_MSG_ERROR([--enable-demangling specified but libiberty not found])
+         fi
+      ])
+
+      if test "x$have_libiberty" = xyes; then
+         AC_CHECK_HEADERS([demangle.h], [], [
+            CPPFLAGS="$CPPFLAGS -I/usr/include/libiberty"
+            AC_CHECK_HEADERS([libiberty/demangle.h], [], [
```

This line should probably work like this
```suggestion
            AC_CHECK_HEADERS([demangle.h], [], [
```

---

I got an error with your suggestion. `AC_CHECK_HEADERS` doesn't find the header.

---

The commit 2104720e4ec91addf510c7198240d3f802bb2c2e is now in `main`. You can use `htop_search_header_dir` for your use case.

```shell
if htop_search_header_dir demangle.h "/usr/include/libiberty"; then
   :
else
   have_libiberty=no
   if test "x$enable_demangling" = xlibiberty; then
      AC_MSG_ERROR([--enable-demangling specified but <demangle.h> not found])
   fi
fi
```